src/tox/venv.py

```diff
@@ -515,6 +515,11 @@ def _pcall(
         env = self._get_os_environ(is_test_command=is_test_command)
         bin_dir = str(self.envconfig.envbindir)
         env["PATH"] = os.pathsep.join([bin_dir, os.environ["PATH"]])
+        if "PATH" in self.envconfig.setenv:
+            env["PATH"] = os.pathsep.join([env["PATH"], self.envconfig.setenv["PATH"]])
+            # Remove redundantly duplicated directories in PATH
+            env["PATH"] = ':'.join(list(set(env["PATH"].split(":"))))
+        os.environ["PATH"] = env["PATH"]
```

Comment:
we should never modify the global path, chaning the env should be all you need to propagate to the subprocess call

Reply:
@gaborbernat if we don't modify the global path, then while doing the `_normal_lookup`, it won't identify the executable. Attaching the snippet for reference
https://github.com/tox-dev/tox/blob/d6cd8d42fa06cc02d747ebe92b3179bc677f0591/src/tox/venv.py#L167-L168

Reply:
that's a bug for the normal lookup though, we should inject the environments generated paths https://py.readthedocs.io/en/latest/path.html#py._path.local.LocalPath.sysfind to fix it